src/engine.ml

```diff
@@ -282,7 +282,11 @@ let incoming_control_client config rng session channel now op data =
                         X509.Certificate.pp ca);
           X509.Authenticator.chain_of_trust ~time:(fun () -> Some now) [ ca ]
       in
-      Tls.(Engine.client (Config.client ~authenticator ()))
+      let certificates =
+        match Config.find Tls_cert config, Config.find Tls_key config with
+        | Some cert, Some (`RSA key) -> `Single ([cert], key)
+        | _ -> `None in
```

Comment:
could we move the `in` to the next line?

on a more general topic, should we combine in the configuration parser the certificate + key into a single configuration key? this could as well already check that the public key of the certificate is the same as the public key part of the private key (and avoid `Tls.Engine.client` from raising an exception). I don't think one of cert/key without the other is useful.